scripts/backport.py

```diff
@@ -0,0 +1,261 @@
+                request.status_code, query
+            )
+        )
+
+
+# Get the number of issue fixed by the given pull request.
+def get_referenced_issue(pr_number):
+    ref_result = run_query(
+        string.Template(
+            """
+        query {
+            repository(owner: "timescale", name: "timescaledb") {
+              pullRequest(number: $pr_number){
+                closingIssuesReferences(first: 1) {
+                  edges {
+                    node {
+                      number
+                    }
+                  }
+                }
+              }
+            }
+          }
+          """
+        ).substitute({"pr_number": pr_number})
+    )
+
+    # The above returns {'data': {'repository': {'pullRequest': {'closingIssuesReferences': {'edges': [{'node': {'number': 4944}}]}}}}}
+
+    ref_edges = ref_result["data"]["repository"]["pullRequest"][
+        "closingIssuesReferences"
+    ]["edges"]
+
+    if ref_edges:
+        return ref_edges[0]["node"]["number"]
+
+    return None
+
+
+# The token has to have the "access public repositories" permission, or else creating a PR returns 404.
+github = Github(os.environ.get("GITHUB_TOKEN"))
+source_repo = github.get_repo("timescale/timescaledb")
+source_remote = "origin"
+target_repo_name = os.environ.get("BACKPORT_TARGET_REPO")
+target_remote = os.environ.get("BACKPORT_TARGET_REMOTE")
+if not target_repo_name or not target_remote:
+    print(
+        "Please specify the BACKPORT_TARGET_REPO (e.g. `timescale/timescaledb`) and BACKPORT_TARGET_REMOTE (e.g. `origin`) environment variables.",
+        file=sys.stderr,
+    )
+    exit(1)
+
+target_repo = github.get_repo(target_repo_name)
+
+# Fetch the sources
+subprocess.run(f"git fetch {source_remote}", shell=True).check_returncode()
+subprocess.run(f"git fetch {target_remote}", shell=True).check_returncode()
+
+# Find out what is the branch corresponding to the previous version compared to
+# main. We will backport to that branch.
+version_config = dict(
+    [
+        (x[0].strip(), x[1].strip())
+        for x in [
+            l.split("=")
+            for l in subprocess.check_output(
+                f"git show {source_remote}/main:version.config",
+                shell=True,
+                text=True,
+            ).splitlines()
+            if l
+        ]
+    ]
+)
+
+previous_version = version_config["update_from_version"]
+previous_version_parts = previous_version.split(".")
+previous_version_parts[-1] = "x"
+backport_target = ".".join(previous_version_parts)
+
+print(f"Will backport to {backport_target}")
+
+# Find out which commits are unique to main and target branch. Also build sets of
+# the titles of these commits. We will compare the titles to check whether a
+# commit was backported.
+main_commits = [
+    l.split("\t")
+    for l in subprocess.check_output(
+        f'git log -1000 --pretty="format:%h\t%s" {source_remote}/2.9.x..{source_remote}/main',
```

Comment:
Could we make the branch name dynamic (here and in Line 123)?

Reply:
Thanks, fixed.